test/Microsoft.Health.Fhir.Shared.Tests.Crucible/CrucibleTestFixture.cs

```diff
@@ -0,0 +1,104 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Health.Fhir.Tests.Common;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Crucible
+{
+    public class CrucibleTestFixture : IClassFixture<CrucibleDataSource>
+    {
+        private readonly CrucibleDataSource _dataSource;
+        private readonly ITestOutputHelper _output;
+
+        public CrucibleTestFixture(CrucibleDataSource dataSource, ITestOutputHelper output)
+        {
+            _dataSource = dataSource;
+            _output = output;
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTests))]
+        [Trait(Traits.Category, Categories.Crucible)]
+        public void Run(string id)
+        {
+            // IEnumerable<object> testIds;
+
+            if (string.IsNullOrEmpty(id))
+            {
+                // testIds = CrucibleTestFixture.GetTests();
+                return;
+            }
+
+            var findTest = _dataSource.TestRun.Value.TestRun.TestResults.FirstOrDefault(x => x.TestId == id);
+
+            if (findTest != null)
+            {
+                var failures = findTest.Result
+                    .Where(x =>
+                    {
+                        var testName = $"{x.TestId ?? findTest.TestId}/{x.Id}";
+                        return x.Status == "fail" && !KnownCrucibleTests.KnownFailures.Contains(testName) && !KnownCrucibleTests.KnownBroken.Contains(testName)
+                            && !x.Message.ToString().Contains("Search bundle returned does not report a total entry count");
```

Comment:
I think it would be better to keep the logic in here generic. Maybe KnownCrucibleTests.TestFilters?